File: packages/implementations/OneLang-File-v0.1/PHP/src/OneFile.php
<?php

namespace OneLang\File;

class OneFile {
    static function listFiles($dir, $recursive) {
        if (!$recursive)
            throw "Not supported";
        if (substr($dir, -1) !== "/")
            $dir .= "/";

        $rii = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($dir));

        $files = array(); 
        foreach ($rii as $file)
            if (!$file->isDir())
                $files[] = substr($file->getPathname(), strlen($dir));

        sort($files);
        return $files;
    }

    static function readText($fn) {
        return file_get_contents($fn);
    }

    static function providePath($fn) { 
        $dir = dirname($fn);
        if (!file_exists($dir))
            mkdir($dir, 0777, true);
        return $fn;
    }

    static function writeText($fn, $contents) {
        //print("writing file: $fn\n");
        file_put_contents(OneFile::providePath($fn), $contents);
    }

    static function copy($src, $dst) {
        copy($src, OneFile::providePath($dst));
    }
}

File: packages/implementations/OneLang-File-v0.1/Python/src/OneFile.py
import pathlib
import os
import shutil

class OneFile:
    def read_text(fn):
        with open(fn, newline='') as f: return f.read()
    
    def write_text(fn, content):
        os.makedirs(os.path.dirname(fn), exist_ok=True)
        with open(fn, 'w') as f: f.write(content)

    def list_files(dir, recursive):
        files = list(sorted([str(x.relative_to(dir)) for x in pathlib.Path(dir).glob("**/*") if x.is_file()]))
        return files

    def copy(src, dst):
        os.makedirs(os.path.dirname(dst), exist_ok=True)
        shutil.copy(src, dst)

